multicast/src/test/kotlin/com/dropbox/flow/multicast/MulticastTest.kt

```diff
@@ -398,6 +399,34 @@ class MulticastTest {
         unlockC1.complete(Unit)
     }
 
+    @Test
+    fun closed() = testScope.runBlockingTest {
+        var collectionCount = 0
+        val multicaster = createMulticaster(flow {
+           collectionCount ++
+            emit(1)
+            // suspend forever
+            suspendCancellableCoroutine<Unit> {}
+        })
+        val collection = async {
+            multicaster.flow.toList()
+        }
+        runCurrent()
+        assertThat(collection.isActive).isTrue()
+        multicaster.close()
+        runCurrent()
+        assertThat(collection.isCompleted).isTrue()
+        assertThat(collection.await()).isEqualTo(listOf(1))
+
+        // now add a new subscriber, should just close immediately
+        val collection2 = async {
+            multicaster.flow.toList()
+        }
+        runCurrent()
+        assertThat(collection2.isActive).isFalse()
+        assertThat(collection2.await()).isEmpty()
```

Comment:
1) there are 2 tests here. maybe we can split them into 2 tests so if one fails we know exactly what's wrong
2) it would be nice to also add a test to verify expected behavior with buffering enabled + close + a 2nd channel comes in. should a 2nd channel get a value or not (I'm guessing yes?)

Reply:
no, second channel does not get any value.
basically close cleans up all resources and makes it unusable.
This section of the test asserts exactly that (minus buffer),
see:
`// now add a new subscriber, should just close immediately` 

i'll move this section into another test